### PR TITLE
Accelerating disjoint set computation on GPU.

### PR DESCRIPTION
**Pull request by yatisht**

### CODE REVIEW

File: src/Assembler.hpp [hunk 1]
```diff
@@ -285,6 +285,39 @@ class shasta::Assembler :
...
         size_t threadCount
     );
+    
+    // Loop over all alignments in the read graph
+    // to create vertices of the global marker graph.
+    // Throw away vertices with coverage (number of markers)
+    // less than minCoverage or more than maxCoverage.
+    // Also throw away "bad" vertices - that is, vertices
+    // with more than one marker on the same oriented read.
+    void createMarkerGraphVerticesGpu(
```
Comment (paoloczi) [inline src/Assembler.hpp]: I would prefer to put this declaration under `#ifdef SHASTA_BUILD_FOR_GPU`. I know it does not hurt, but it is just for readability, to make it clear that this function is only compiled in in that case.

File: src/gpu/AssemblerAlign-gpu.cpp [hunk 2]
```diff
@@ -124,8 +124,8 @@ void Assembler::computeAlignmentsGpu(
     cout << timestamp << "Creating alignment table." << endl;
     computeAlignmentTable();
 
-    cout << timestamp << "Shutting down processors." << endl;
-    shasta_shutdownProcessors(nDevices);
+    //cout << timestamp << "Shutting down processors." << endl;
```
Comment (paoloczi) [inline src/gpu/AssemblerAlign-gpu.cpp]: Fine to leave the message commented out, but nevertheless I would make it less ambiguous by changing "processors" to "GPUs" or something else. Most people would normally interpret "processors" as meaning "CPUs".

File: src/gpu/AssemblerAlign-gpu.cpp [hunk 3]
```diff
@@ -124,8 +124,8 @@ void Assembler::computeAlignmentsGpu(
...
+    //shasta_shutdownProcessors(nDevices);
```
Comment (paoloczi) [inline src/gpu/AssemblerAlign-gpu.cpp]: Why are we not shutting down the processors?

**Comment by paoloczi** [comment]
I made some small comments that you can address at your convenience, but  I will merge the PR anyway.

File: src/gpu/AssemblerAlign-gpu.cpp (hunk 3 above)
Comment (yatisht) [reply src/gpu/AssemblerAlign-gpu.cpp]: They are now shut down later on (after disjoint set computation).

**Pull request merged**